### PR TITLE
feat: read test cases dynamically from a text file

### PR DESCRIPTION
**Pull request by BeanieMen**
meow

### CODE REVIEW

**Comment by BeanieMen** [comment]
@nopeless request for review

File: test/tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
       "@/*": [
         "../src/*"
       ]
-    }
+    },
+    "module": "NodeNext"
```
Comment (nopeless) [inline test/tsconfig.json]: is this required @BeanieMen

File: src/lib/util.ts [hunk 2]
```diff
@@ -236,8 +236,9 @@ export function indentTrailing(
     .join(`\n`);
 }
 
-export function readFileLines(filePath: string) {
-  return readFileSync(filePath, `utf-8`).split(/\r?\n/);
+export async function readFileLines(filePath: string) {
+  const data = await readFile(filePath, `utf-8`)
+  return data.split(/\r?\n/);
```
Comment (nopeless) [inline src/lib/util.ts]: Can you ignore lines starting with ;? (ini style comments)

**Comment by BeanieMen** [comment]
I don't see any reviews

**Review by nopeless — CHANGES_REQUESTED** [review]

File: test/tsconfig.json (hunk 1 above)
Comment (BeanieMen) [reply test/tsconfig.json]: Yes, i was getting a error saying if module resolution is node next, this must also be node next

File: src/lib/util.ts (hunk 2 above)
Comment (BeanieMen) [reply src/lib/util.ts]: aight

File: test/tsconfig.json (hunk 1 above)
Comment (BeanieMen) [reply test/tsconfig.json]: ```Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'```

Comment (nopeless) [reply test/tsconfig.json]: what gave you that error?

**Comment by nopeless** [comment]
lgtm

**Pull request merged**